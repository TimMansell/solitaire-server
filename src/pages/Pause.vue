<template>
  <GameOverlay center-content show-logo data-test="game-paused">
    <template #title> Game Paused page</template>
    <template #msg>
      <p v-if="!isGameActive">Your game has been paused due to inactivity</p>
      <GameSummary />
    </template>
    <template #buttons>
      <ResumeGameButton />
    </template>
  </GameOverlay>
</template>

<script>
import { mapGetters } from 'vuex';
import GameOverlay from '@/components/GameOverlay.vue';
import GameSummary from '@/components/GameSummary.vue';
import ResumeGameButton from '@/components/ResumeGameButton.vue';

export default {
  name: 'GamePaused',
  components: {
    GameOverlay,
    ResumeGameButton,
    GameSummary,
  },
  computed: {
    ...mapGetters(['isGameActive']),
  },
};
</script>
