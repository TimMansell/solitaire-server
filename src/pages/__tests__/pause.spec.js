import { shallowMount } from '@vue/test-utils';
import Pause from '@/pages/Pause.vue';

describe('Pause.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(Pause);

    expect(wrapper).toMatchSnapshot();
  });
});
