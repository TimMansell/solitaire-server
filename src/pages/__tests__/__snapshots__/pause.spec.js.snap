// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pause.vue matches snapshot 1`] = `
<gameoverlay-stub centercontent="true" showlogo="true" data-test="game-paused"> Game Paused<gamesummary-stub></gamesummary-stub>
  <resumegamebutton-stub></resumegamebutton-stub>
</gameoverlay-stub>
`;
